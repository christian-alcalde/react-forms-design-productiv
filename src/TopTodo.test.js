import React from "react";
import { render } from "@testing-library/react";
import TopTodo from "./TopTodo";

const testTodos = [
  {
    id: "1",
    title: "laundry",
    description: "testDescription",
    priority: 3,
  },
  {
    id: "2",
    title: "water plants",
    description: "testDescription2",
    priority: 1,
  },
  {
    id: "3",
    title: "fix bike",
    description: "testDescription3",
    priority: 2,
  },
];

it("renders without crashing", function () {
  render(<TopTodo todos={testTodos} />);
});

it("matches snapshot", function () {
  const { container } = render(<TopTodo todos={testTodos} />);
  expect(container).toMatchSnapshot();
});

it("shows oldest todo with highest priority", function () {
  const { container } = render(<TopTodo todos={testTodos} />);

  expect(container).toHaveTextContent("water plants");
  expect(container).not.toHaveTextContent("laundry");
  expect(container).not.toHaveTextContent("fix bike");
});
