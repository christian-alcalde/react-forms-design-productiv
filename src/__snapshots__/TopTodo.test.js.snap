// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        water plants
      </b>
       
      <small>
        (Priority:
        1
        )
      </small>
    </div>
    <div>
      <small>
        testDescription2
      </small>
    </div>
  </div>
</div>
`;
